Core/Business/Controller/Admin/ProductController.php

```diff
@@ -0,0 +1,361 @@
+use PrestaShop\PrestaShop\Core\Foundation\Controller\BaseController;
+use PrestaShop\PrestaShop\Core\Business\Form\Type\ChoiceCategoriesTreeType;
+use PrestaShop\PrestaShop\Adapter\Product\AdminProductDataProvider;
+use PrestaShop\PrestaShop\Core\Foundation\Exception\ErrorException;
+use PrestaShop\PrestaShop\Core\Foundation\Exception\DevelopmentErrorException;
+use PrestaShop\PrestaShop\Core\Foundation\Exception\WarningException;
+
+/**
+ * Admin controller for the Product pages using the Symfony architecture:
+ * - categories
+ * - product list
+ * - product details
+ * - product attributes
+ * - ...
+ *
+ * This controller is the first one to be beta tested on the new Symfony Architecture.
+ * The retro-compatibility is dropped for the corresponding Admin pages.
+ * A set of hooks are integrated and an Adapter is made to wrap the new EventDispatcher
+ * component to the existing hook system. So existing hooks are always triggered, but from the new
+ * code (and so needs to be adapted on the module side ton comply on the new parameters formats, the new UI style, etc...).
+ *
+ * FIXME: to adapt after 1.7.0 when alternative behavior will be removed (@see self::shouldUseLegacyPagesAction()).
+ */
+class ProductController extends AdminController
+{
+    use AutoObjectInflaterTrait; // auto inflate objects if pattern found in the route format.
+    use AutoResponseFormatTrait; // try to auto fill template engine parameters according to the current action.
+    use SfControllerResolverTrait; // dependency injection in sf way.
+
+    /**
+     * Get the Catalog page with stats banner, product list, bulk actions, filters, search, etc...
+     *
+     * URL example: /product/catalog/40/20/id_product/asc
+     *
+     * @param Request $request
+     * @param Response $response
+     */
+    public function productCatalogAction(Request &$request, Response &$response)
+    {
+        // Redirect to legacy controller (FIXME: temporary behavior)
+        if ($this->shouldUseLegacyPages()) {
+            $this->redirectToRoute(
+                $request,
+                'admin_product_catalog',
+                array(
+                    // do not tranmit limit & offset: go to the first page when
+                    'productOrderby' => $request->attributes->get('orderBy'),
+                    'productOrderway' => $request->attributes->get('orderWay')
+                ),
+                true, // force legacy URL
+                false // temporary
+            );
+        }
+
+        // Retrieve persisted filter parameters
+        $dataProvider = $this->container->make('CoreAdapter:Product\\AdminProductDataProvider');
+        $persistedFilterParameters = $dataProvider->getPersistedFilterParameters('ls_products_');
+        $persistedFilterParameters = array_replace($persistedFilterParameters, $request->request->all());
+        $response->addContentData(null, $persistedFilterParameters);
```

Comment:
Not sure to understand the behavior intended here ? Can you give me some details @xGouley please ? Thx !